autoload/lsp/options.vim

```diff
@@ -53,6 +53,25 @@ export var lspOptions: dict<any> = {
   # icase | fuzzy | case match for language servers that replies with a full
   # list of completion items
   completionMatcher: 'case',
+  # diagnostics signs options
+  diagSignOpts: {
+      error: {
+          text: 'E>',
+          texthl: 'ErrorMsg',
+      },
+      warning: {
+          text: 'W>',
+          texthl: 'Search',
+      },
+      info: {
+          text: 'I>',
+          texthl: 'Pmenu',
+      },
+      hint: {
+          text: 'H>',
+          texthl: 'Question',
+      },
+  },
 }
```

Comment:
It's worth to keep in mind that using nested objects as options, will force the user to set all properties as once, as otherwise some properties will be removed, eg:

    g:LspOptionsSet({ diagSignOpts: { error: { text: 'E~', texthl: 'ErrorMsg' } } }) 

will result in:

 ```
  diagSignOpts: {
      error: {
          text: 'E~',
          texthl: 'ErrorMsg',
      },
  },
  ```

It might be worth mentioning it in the help text, or change the code to have many options:

    diagSignErrorText: 'E>',
    diagSignErrorTexthl: 'ErrorMsg',
    ...

Reply:
Thank you for comments I change options to separate